src/MasterTable.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react';
+import { httpClient } from "./handlers/axiosConfig"
+import Cookies from 'universal-cookie'
+import ReactDataGrid from "react-data-grid";
+import { ClipLoader } from 'react-spinners';
+/**
+ * Master table of all volunteers
+ */
+const columns = [
+    { key: "email", name: "Email"},
+    { key:"first_name", name:"First Name" },
+    { key:"current_action", name:"Current Action", editable:true}
+];
+export default class MasterTable extends Component {
+
+    constructor(props) {
+        super(props)
+        this.state = {
+            data: [],
+            loading: false
+        }
+    }
+    //fetch data on mounting
+    componentDidMount() {
+        const cookie = new Cookies();
+        const request_data = {
+            "email": cookie.get("email"),
+            "token": cookie.get("auth")
+        }
+        this.setState({
+            loading: true
+        })
+        httpClient.post("/getvolunteers", request_data).then(response => {
```

Comment:
You seem to assume a lot about the backend. Is the api documented somewhere?

Reply:
i'll add comments in a coming PR that document the params for the backend